x/staking/keeper/grpc_query.go

```diff
@@ -0,0 +1,460 @@
+package keeper
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// Querier is used as Keeper will have duplicate methods if used directly, and gRPC names take precedence over keeper
+type Querier struct {
+	Keeper
+}
+
+var _ types.QueryServer = Querier{}
+
+// Validators queries all validators that match the given status
```

Comment:
afaik you can also query for all validators by not providing the status

Reply:
yes, but I was trying not to disturb existing querier format which filters validators by status